src/primitives/transaction.h

```diff
@@ -345,6 +345,12 @@ class CTxOut
     {
         return a.scriptPubKey < b.scriptPubKey;
     }
+
+    // ELEMENTS: a "confidential" output has commitments instead of explicit values
+    bool IsConfidential() const
+    {
+        return nValue.IsCommitment() && nAsset.IsCommitment() && nNonce.IsCommitment();
```

Comment:
In d0c18edda0274dab7d8582f521e9bd86ccd85925:

This is okay, but I really think it'd make more conceptual sense to treat the individual fields individually, instead of taking away the CT discount unless every field is present and confidential.

Reply:
done in a7e9d931ec9065b5da3c271660dc8f91b6c1bd29